pkg/analysis/passes/binarypermissions/binarypermissions_test.go

```diff
@@ -0,0 +1,123 @@
+package binarypermissions
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/grafana/plugin-validator/pkg/analysis"
+	"github.com/grafana/plugin-validator/pkg/analysis/passes/archive"
+	"github.com/grafana/plugin-validator/pkg/analysis/passes/metadata"
+	"github.com/grafana/plugin-validator/pkg/testpassinterceptor"
+	"github.com/stretchr/testify/require"
+)
+
+const pluginId = "test-plugin-panel"
+
+func TestBinaryFoundWithCorrectPermissions(t *testing.T) {
+	var interceptor testpassinterceptor.TestPassInterceptor
+	pluginJsonContent := []byte(`{
+		"id": "` + pluginId + `",
+		"type": "panel",
+		"executable": "test-plugin-panel"
+	}`)
+
+	// due to CI running on linux, we need to re-create the correct permissions
+	testDataContainer := filepath.Join("testdata", "correct-permissions")
+	os.Chmod(filepath.Join(testDataContainer, "test-plugin-panel_linux_amd64"), 0755)
```

Comment:
`os.Chmod` returns an `error`. I would handle it just to be safe, like this:

```suggestion
	require.NoError(t, os.Chmod(filepath.Join(testDataContainer, "test-plugin-panel_linux_amd64"), 0755))
```

Same for the other occurrences

Reply:
That's a nice addition to debug a broken test in case the OS has a different permission system. I'll add it.